src/Tizen.NUI/src/public/WindowSystem/QuickPanelClient.cs

```diff
@@ -0,0 +1,422 @@
+            {
+                throw new ArgumentNullException((string)"win");
+            }
+
+            _tzsh = tzShell;
+            _tzshWin = win.GetNativeId();
+            _tzshQpClient = Interop.QuickPanelClient.CreateWithType(_tzsh.GetNativeHandle(), (IntPtr)_tzshWin, (int)type);
+            if (_tzshQpClient == IntPtr.Zero)
+            {
+                int err = Tizen.Internals.Errors.ErrorFacts.GetLastResult();
+                _tzsh.ErrorCodeThrow(err);
+            }
+        }
+
+        /// <summary>
+        /// Destructor.
+        /// </summary>
+        /// <since_tizen> 8 </since_tizen>
+        ~QuickPanelClient()
+        {
+            Dispose(false);
+        }
+
+        /// <summary>
+        /// Dispose.
+        /// </summary>
+        /// <exception cref="ArgumentException">Thrown when failed of invalid argument.</exception>
+        /// <since_tizen> 8 </since_tizen>
+        public void Dispose()
+        {
+            Dispose(true);
+            GC.SuppressFinalize(this);
+        }
+
+        /// <inheritdoc/>
+        protected virtual void Dispose(bool disposing)
+        {
+            int _res;
+            if (!disposed)
+            {
+                if (_tzshQpClient != IntPtr.Zero)
+                {
+                    _res = Interop.QuickPanelClient.Destroy(_tzshQpClient);
+                    _tzsh.ErrorCodeThrow(_res);
+                    _tzshQpClient = IntPtr.Zero;
+                }
+                disposed = true;
+            }
+        }
+
+        /// <exception cref="ArgumentException" > Thrown when failed of invalid argument.</exception>
+        private void OnVisibleChanged(int type, IntPtr ev_info, IntPtr data)
+        {
+            int _res;
+            int _state;
+
+            _res = Interop.QuickPanelClient.GetEventVisible(ev_info, out _state);
+            _tzsh.ErrorCodeThrow(_res);
+
+            _visibleChanged?.Invoke(this, (StateVisible)_state);
+        }
+
+        /// <exception cref="ArgumentException" > Thrown when failed of invalid argument.</exception>
+        private void OnOrientationChanged(int type, IntPtr ev_info, IntPtr data)
+        {
+            int _res;
+            int _state;
+
+            _res = Interop.QuickPanelClient.GetEventOrientation(ev_info, out _state);
+            _tzsh.ErrorCodeThrow(_res);
+
+            _orientationChanged?.Invoke(this, (StateOrientation)_state);
+        }
+
+        /// <summary>
+        /// Gets the visible state of the quickpanel.
+        /// </summary>
+        /// <returns>The visible state of the quickpanel service window.</returns>
+        /// <exception cref="ArgumentException" > Thrown when failed of invalid argument.</exception>
+        /// <exception cref="InvalidOperationException">Thrown when failed because of an invalid operation or no service.</exception>
+        /// <since_tizen> 8 </since_tizen>
+        public StateVisible GetVisible()
+        {
+            int _res;
```

Comment:
Don't use uderscore(_) for local method variables. It only allows for private class members.

Reply:
I removed all of uderscore in local method variables.